facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -0,0 +1,446 @@
+#' @param aggregations Data frame with columns `name`, `var_weight`, `metric`,
+#'   `group_by`, `compute_fn`, `post_fn`. Each row represents one aggregate
+#'   to report. `name` is the aggregate's base column name; `var_weight` is the 
+#'   column to use for its weights; `metric` is the column of `df` containing the
+#'   response value. `group_by` is a list of variables used to perform the 
+#'   aggregations over. `compute_fn` is the function that computes
+#'   the aggregate response given many rows of data. `post_fn` is applied to the
+#'   aggregate data after megacounty aggregation, and can perform any final
+#'   calculations necessary.
+#' @param col_var Name of response var
+#'
+#' @return list of data frame of individual response data and user-set data 
+#' frame of desired aggregations
+#'
+#' @export
+code_binary <- function(df, aggregations, col_var) {
+  df <- remap_response(df, col_var, c("1"=1, "2"=0, "3"=NA))
+  return(list(df, aggregations))
+}
+
+#' Convert a single multi-select response column to a set of boolean columns
+#' 
+#' Update aggregations table to use new set of columns where `col_var` had
+#' previously been used as the metric to aggregate. Does not change columns
+#' referenced in `groupby`
+#' 
+#' @param df Data frame of individual response data.
+#' @param aggregations Data frame with columns `name`, `var_weight`, `metric`,
+#'   `group_by`, `compute_fn`, `post_fn`. Each row represents one aggregate
+#'   to report. `name` is the aggregate's base column name; `var_weight` is the 
+#'   column to use for its weights; `metric` is the column of `df` containing the
+#'   response value. `group_by` is a list of variables used to perform the 
+#'   aggregations over. `compute_fn` is the function that computes
+#'   the aggregate response given many rows of data. `post_fn` is applied to the
+#'   aggregate data after megacounty aggregation, and can perform any final
+#'   calculations necessary.
+#' @param col_var Name of response var
+#'
+#' @return list of data frame of individual response data and user-set data 
+#' frame of desired aggregations
+#' 
+#' @importFrom stats na.omit
+#' @importFrom tibble add_row
+#' @importFrom stringi stri_replace_all
+#'
+#' @export
+code_multiselect <- function(df, aggregations, col_var) {
+  # Get unique response codes. Sort alphabetically.
+  response_codes <- sort( na.omit(
+    unique(do.call(c, strsplit(unique(df[[col_var]]), ",")))))
+  
+  # Turn each response code into a new binary col
+  new_binary_cols <- as.character(lapply(
+    response_codes, 
+    function(code) {
+      paste(col_var, 
+            stri_replace_all(code, "_", fixed=" "), 
+            sep="_")
+      }
+    ))
+  #### TODO: eval(parse()) here is not the best approach, but I can't find another 
+  # way to get col_var (a string) to be used as a var that references a column
+  # rather than as an actual string. This approach causes a shallow copy to be 
+  # made (warning is raised).
```

Comment:
You probably want one of these:
```
!!sym(col_var)
!!as.symbol(col_var)
!!as.name(col_var)
```

Reply:
I can't get the `!!` operator to be recognized in this context. I can't quite find a clear explanation why, but it seems like it's something to do with `!!` only being recognized within `dplyr` functions, whereas this is in a regex function (but within a `dplyr` function...). I'm going to shelve this for now and revisit as an incremental update.